lib/galaxy/webapps/galaxy/api/object_store.py

```diff
@@ -0,0 +1,80 @@
+"""
+API operations on Galaxy's object store.
+"""
+import logging
+from typing import (
+    Any,
+    Dict,
+    List,
+)
+
+from fastapi import (
+    Path,
+    Query,
+)
+
+from galaxy.exceptions import (
+    ObjectNotFound,
+    RequestParameterInvalidException,
+)
+from galaxy.managers.context import ProvidesUserContext
+from galaxy.objectstore import BaseObjectStore
+from . import (
+    depends,
+    DependsOnTrans,
+    Router,
+)
+
+log = logging.getLogger(__name__)
+
+router = Router(tags=["object sstore"])
+
+ConcreteObjectStoreIdPathParam: str = Path(
+    ..., title="Concrete Object Store ID", description="The concrete object store ID."
+)
+
+SelectableQueryParam: bool = Query(
+    default=False, title="Selectable", description="Restrict index query to user selectable object stores."
+)
+
+
+@router.cbv
+class FastAPIObjectStore:
+    object_store: BaseObjectStore = depends(BaseObjectStore)
+
+    @router.get(
+        "/api/object_store",
+        summary="",
+        response_description="",
+    )
+    def index(
+        self,
+        trans: ProvidesUserContext = DependsOnTrans,
+        selectable: bool = SelectableQueryParam,
+    ) -> List[Dict[str, Any]]:
+        if not selectable:
+            raise RequestParameterInvalidException(
+                "The object store index query currently needs to be called with selectable=true"
+            )
+        selectable_ids = self.object_store.object_store_ids_allowing_selection()
+        return [self._dict_for(selectable_id) for selectable_id in selectable_ids]
+
+    @router.get(
+        "/api/object_store/{object_store_id}",
+        summary="Return boolean to indicate if Galaxy's default object store allows selection.",
```

Comment:
It seems like this should work for all object stores, and it also doesn't return a boolean.

Reply:
Fixed in the follow PR already I think.